benchmark/benchmark.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+require_once __DIR__ . '/shared.php';
+
+$commitResult = ($argv[1] ?? 'false') === 'true';
+$phpCgi = $argv[2] ?? dirname(PHP_BINARY) . '/php-cgi';
+if (!file_exists($phpCgi)) {
+    fwrite(STDERR, "php-cgi not found\n");
+    exit(1);
+}
+
+function main() {
+    global $commitResult;
+
+    $data = [];
+    $data['bench.php'] = runBench();
+    $data['symfony_demo'] = runSymfonyDemo();
+    $data['wordpress_6.2'] = runWordpress();
+    $result = json_encode($data, JSON_PRETTY_PRINT) . "\n";
+
+    if ($commitResult) {
+        commitResult($result);
+    } else {
+        fwrite(STDOUT, $result);
+    }
+}
+
+function commitResult(string $result) {
+    $repo = __DIR__ . '/repos/data';
+    cloneRepo($repo, 'git@github.com:iluuu1994/php-benchmark-data.git');
+    runCommand(['git', 'pull', '--end-of-options', 'origin'], $repo);
+
+    $commitHash = getPhpSrcCommitHash();
+    $dir = $repo . '/' . substr($commitHash, 0, 2) . '/' . $commitHash;
+    $summaryFile = $dir . '/summary.json';
+    if (!is_dir($dir)) {
+        mkdir($dir, 0755, true);
+    }
+    file_put_contents($summaryFile, $result);
+
+    runCommand(['git', 'add', '--end-of-options', $summaryFile], $repo);
+    runCommand(['git', 'commit', '--allow-empty', '-m', 'Add result for php/php-src@' . $commitHash, '--author', 'Benchmark <benchmark@php.net>'], $repo);
+    runCommand(['git', 'push'], $repo);
+}
+
+function getPhpSrcCommitHash(): string {
+    $result = runCommand(['git', 'log', '--pretty=format:%H', '-n', '1'], dirname(__DIR__));
+    return $result->stdout;
+}
+
+function runBench(): array {
+    $process = runValgrindPhpCgiCommand([dirname(__DIR__) . '/Zend/bench.php']);
+    return ['instructions' => extractInstructionsFromValgrindOutput($process->stderr)];
+}
+
+function runSymfonyDemo(): array {
+    $dir = __DIR__ . '/repos/symfony-demo-2.2.3';
+    cloneRepo($dir, 'git@github.com:iluuu1994/symfony-demo-2.2.3.git');
+    runPhpCommand([$dir . '/bin/console', 'cache:clear']);
+    runPhpCommand([$dir . '/bin/console', 'cache:warmup']);
+    $process = runValgrindPhpCgiCommand(['-T1,1', $dir . '/public/index.php']);
+    return ['instructions' => extractInstructionsFromValgrindOutput($process->stderr)];
+}
+
+function runWordpress(): array {
+    $dir = __DIR__ . '/repos/wordpress-6.2';
+    cloneRepo($dir, 'git@github.com:iluuu1994/wordpress-6.2.git');
+
+    /* FIXME: It might be better to use a stable version of PHP for this command because we can't
+     * easily alter the phar file */
+    runPhpCommand([
+        '-d error_reporting=0',
+        'wp-cli.phar', 
+        'core',
+        'install',
+        '--url=wordpress.local',
+        '--title="Wordpress"',
+        '--admin_user=wordpress',
+        '--admin_password=wordpress',
+        '--admin_email=benchmark@php.net',
+    ], $dir);
+
+    // Warmup
+    runPhpCommand([$dir . '/index.php'], $dir);
```

Comment:
Does warmup imply compiling files in opcache? If yes, shouldn't the compiling be benchmarked as well?

Reply:
We generally care less about compilation time than runtime as you should use opcache, in which case compilation is skipped. This PR only adds the basics, more benchmarks for compilation itself in the future are conceivable.

Reply:
But no, this call isn't about opcache. It's about making sure wordpress generates any cache files that might be required. I don't know wordpress at all so this might be senseless. But it's supposed to be the equivalent to the `cachw:warmup` in Symfony.

Reply:
@jrfnl Could you please double check the Wordpress config?

Reply:
> I presume the committed version of WP 6.2 is based on the distribution zip ?

Indeed.

> As far as I know, WP doesn't natively generate cache files.

Good to know! I noticed that the first request is significantly slower. Maybe it's database stuff? Either way, I don't think this call really hurts as it only takes a second.

Reply:
> I noticed that the first request is significantly slower. 

I wonder if that is the opcache priming ? As WP doesn't use autoloading at all (with the exception of some included external dependencies), any request will `include`/`require` lots of files, even when not needed.

Reply:
@jrfnl Opcache shm is discarded when the process ends, so probably not.